### DISCUSSION TITLE
[Question]: Generating dynamic mask using API and Node-Red?

### DISCUSSION THREAD

**Discussion by wgregorian** [blakeblackshear/frigate#11038] [Ask A Question] [answered]
### What is your question:

Hi, 

I use the Node-Red MQTT-in node to search for events; one of those happens to be that one parked car that, over time, becomes a stationary object in the events. I want to record all events, but I would like to mask the vehicle on the street that becomes stationary dynamically.

I want to prevent the repeated recording of the same object in the same box. The box coordinates always remain the same. I aim to write a flow that evaluates the event if it repeats N times, then sends an HTTP request to the Frigate API to mask the car object on subsequent detections/recordings until the object is gone, and that's to preserve disk space and improve the signal-to-noise factor.

Am I overthinking it, and is there a more elegant way to handle this?

**Comment by NickM-27** [comment]
what is your end goal exactly? You can already configure frigate to not record stationary objects. You can also already tell if an object is stationary by looking at the `stationary` field in the MQTT /events payload

**Reply by wgregorian** [reply]
If I mask, then I'm not capturing anything else. I want to capture that stationary object's first/last event; ignoring the rest as noise. I thought I would rely on the conditions, such as the box coordinates and the `stationary` Boolean, to create the mask for the object after the first event and remove it after the last using the API.

Or can I use node-red to tell Frigate, "Stop recording that thing that becomes stationary"? 

The end goal is to avoid a scenario like this one: 
![Screenshot 2024-04-19 102006](https://github.com/blakeblackshear/frigate/assets/1276347/7aeb21fe-6fff-4541-b099-4b786a567b76)

**Reply by NickM-27** [reply, marked as answer]
We've made a lot of improvements to stationary tracking and stationary objects for the next version of frigate (0.14). Frigate doesn't support totally dynamic masks like that

**Reply by wgregorian** [reply]
Thank you. It's not like storage is expensive; I have plenty of it. I just thought I'd take a stab at it, but I can also live with this and wait for the next version. I appreciate your speedy response!
